dsgrid/project.py

```diff
@@ -409,3 +404,38 @@ def get_myear_column(dataset_id):
                 self._config,
             )
             context.serialize_dataset_metadata_to_file(dataset.dataset_id, metadata_file)
+
+    def _compute_dataset_hash_and_serialize(
+        self, context: QueryContext, cached_datasets_dir: Path, dataset_id: str
+    ) -> Path:
+        """Create a hash that can be used to identify whether the mapping of the dataset to
+        project dimensions can be skipped based on a previous query.
+
+        If a directory with the hash does not already exist, create it and serialize the content
+        used to create the hash.
+
+        Examples of changes that will invalidate the query:
+          - Bump to project major version number
+          - Change to a dataset version
+          - Change to a project's dimension requirements for a dataset
```

Comment:
Where is this captured? In the project major version number?

Reply:
It _should_ be captured by the project major version number. In case someone messes up, though, it's caught more explicitly by line 431. That pulls the `InputDatasetModel` from the project config which includes dimension requirements and dimension mappings.